### ISSUE TITLE
Running this on Openhab 3.0.2

### ISSUE THREAD

**Issue by alxy** [BoBiene/Wolf-Smartset-to-openhab#8]
Hi,

first of all thanks for the cool tool that I just wanted to try with our new heating system and Openhab. Unfortunately, it seems there has been a new major version of Openhab since this tool has been created.

When loading the generated files in the openhab configuration folder I get some new items created (so this seems to work), however, I get the following errors when it tried to parse/apply the rules:
```
2021-04-25 19:00:18.163 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'wolf_smartset.items'
2021-04-25 19:00:39.584 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'wolf_smartset.rules'
2021-04-25 19:04:52.540 [INFO ] [el.core.internal.ModelRepositoryImpl] - Validation issues found in configuration model 'wolf_smartset.rules', using it anyway:
The field Tmp_wolf_smartsetRules.tokenTime refers to the missing type Object
The field Tmp_wolf_smartsetRules.tokenTime refers to the missing type Object
2021-04-25 19:08:22.168 [WARN ] [e.runtime.internal.RuleContextHelper] - Variable 'tokenTime' on rule file 'wolf_smartset.rules' cannot be initialized with value 'org.eclipse.xtext.xbase.impl.XConstructorCallImplCustom@1c81e46 (invalidFeatureIssueCode: null, validFeature: false, explicitConstructorCall: true, anonymousClassConstructorCall: false)': An error occurred during the script execution: null
2021-04-25 19:08:22.359 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID 'wolf_smartset-1' failed: Unknown variable or command '<'; line 9, column 8, length 29 in wolf_smartset
2021-04-25 19:08:22.478 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID 'wolf_smartset-1' failed: Unknown variable or command '<'; line 9, column 8, length 29 in wolf_smartset
2021-04-25 19:08:22.544 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID 'wolf_smartset-1' failed: Unknown variable or command '<'; line 9, column 8, length 29 in wolf_smartset
```

By any chance, do you know how to resolve this?
I have no prior experience with Openhab so its pretty hard for me to tell where the error in the end comes from, but I expect the syntax for the rules has changed a bit between the versions?

**Comment by BoBiene** [comment]
@alxy that's correct currently only OH 2.5.x is supported.

**Comment by alxy** [comment]
Ok, I'll try to look into it and see how difficult it is to make it work with v3. Do you have any good hints to start with, especially on how to test the code. So far I've just modified thje generated rules file and re-uploaded it to the openhab server and watched the logs for any errors. Is that the way to go?

**Comment by BoBiene** [comment]
@alxy the intention is to implement the auth part in the http binding to get rid of the rules and scripting.

If you like you can support this https://github.com/openhab/openhab-addons/pull/9816 as i'm a bit out of time at the moment.

**Comment by alxy** [comment]
Ok, I was able to make that run on OH 3.0 with some modifications. Trying to give a short summary here for anyone interested in doing the modifications: Basically, mainly the `wolf_smartset.rules` file needs to be edited as the syntax for rules changed a bit between the versions.

- [DateTime handling](https://community.openhab.org/t/datetime-conversion-openhab-3-x/107197) is different now. Thus I needed to change the `tokenTime` to `var ZonedDateTime tokenTime = ZonedDateTime.parse("1970-01-01T00:00:00.000Z").withZoneSameInstant(ZoneId.systemDefault())`
- [Date comparison](https://howtodoinjava.com/java/date-time/zoneddatetime-comparison/) needs to be either carried out on the Epoch object, or using the special methods: `if(tokenTime.isBefore(now.plusHours(-1)))`
- [Order of arguments](https://www.openhab.org/docs/configuration/actions.html#exec-actions) for `executeCommandLine` changed. The new syntax looks like this: `executeCommandLine(Duration.ofSeconds(5), "/etc/openhab/scripts/request_config.sh", token, "21309", "27781");`. Note that there are quite a few calls to the function in the generated rules file. All need to be replaced. The old token `@@` isn't valid anymore, all arguments to the script must be passed as comma-separated arguments to the function itself.
- Last but not least: Don't forget to install the Openhab [JsonPath addon](https://www.openhab.org/addons/transformations/jsonpath/). Otherwise the code doesnt run.

Hope these tips help someone :)
